src/App.tsx

```diff
@@ -1,24 +1,203 @@
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const USER_ID = 6550;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(Error.NONE);
+  const [filterBy, setFilterBy] = useState(FilterTodos.ALL);
+  const [activeInput, setActiveInput] = useState(true);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [idUpdating, setIdUpdating] = useState([0]);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        setErrorMessage(Error.NONE);
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setErrorMessage(Error.LOAD);
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const countTodos = todos.length;
+
+  const activeTodos = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const visibleTodos: Todo[] = useMemo(() => {
+    switch (filterBy) {
+      case FilterTodos.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case FilterTodos.COMPLTED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filterBy]);
+
+  const addTodo = async (title: string) => {
+    if (!title.trim()) {
+      setErrorMessage(Error.TITLE);
+
+      return;
+    }
+
+    try {
+      setErrorMessage(Error.NONE);
+
+      const newTodo = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setActiveInput(false);
+      setTempTodo({ ...newTodo, id: 0 });
+      const createdTodo = await createTodo(USER_ID, newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (error) {
+      setErrorMessage(Error.ADD);
+    } finally {
+      setTempTodo(null);
+      setActiveInput(true);
+    }
+  };
+
+  const removeTodo = async (id: number) => {
+    try {
+      setErrorMessage(Error.NONE);
+      setIdUpdating(prevId => [...prevId, id]);
+
+      await deleteTodo(USER_ID, id);
+
+      setTodos(todos.filter(todo => todo.id !== id));
+    } catch (error) {
+      setErrorMessage(Error.DELETE);
+    } finally {
+      setIdUpdating([0]);
+    }
+  };
+
+  const removeCompletedTodos = () => {
+    todos.forEach(async (todo) => {
+      if (todo.completed) {
+        await removeTodo(todo.id);
+      }
+    });
+  };
+
+  const handleUpdate = async (
+    id: number,
+    data: { completed?: boolean, title?: string },
+  ) => {
+    try {
+      setErrorMessage(Error.NONE);
+
+      setIdUpdating(prevId => [...prevId, id]);
+      await updateTodo(USER_ID, id, data);
+
+      setTodos(prevTodos => prevTodos.map(todo => {
+        if (todo.id === id) {
+          const updatedTodo = 'completed' in data ? {
```

Comment:
why do we need this check? why we can't add completed for each todo but with `false` value?

Reply:
not fixed
![image](https://user-images.githubusercontent.com/98764641/227868185-7478c3bc-191e-4083-a153-133fbcf80354.png)

Reply:
because, in data i have or completed or title
so i change it ( if i have completed, i have to set new completed (true or false))